### PR TITLE
Object test

### PR DESCRIPTION
**Pull request by stelisevil**

### CODE REVIEW

File: objects/index.js [hunk 1]
```diff
@@ -76,13 +76,13 @@ export const getObjectValues = (obj) => {
...
+  return Object.assign(objA, objB);
 }
 
 /*
   I expect this function to add a isObject property onto the object passed in:
   isObject should also be set to true
 */
 export const addIsObjectProperty = (obj) => {
-
+  return obj['isObject'] = true; // I cant get this to work so I'm leaving it with what I thought should work
```
Comment (matthewtsinontas) [inline objects/index.js]: You're on the right lines - 

You need to modify the object as you have done - but you are not returning the modified object from this function; you are returning the result from the assignment of a new property to that object

Consider the following if you will...

```js
// here we have a user object
const user = {
  name: 'Matthew',
  age: 24
};
```
This is me modifying the user object
```js
user.isObject = true;
```
The user object now has this new property associated with it
```js
console.log(user.isObject) // true
```
However, this function needs to add the property to the object and then return the new modified object

File: objects/test.js [hunk 2]
```diff
@@ -78,6 +78,8 @@ describe('Object test', () => {
       const newObj = addIsObjectProperty(user);
       expect(newObj).toHaveProperty('isObject');
       expect(newObj.isObject).toEqual(true);
+      expect(newObj).toHaveProperty('firstName');
+      expect(newObj.isObject).toEqual('Matthew');
```
Comment (matthewtsinontas) [inline objects/test.js]: Read the test on line 82, it doesn't quite make sense 😅

File: objects/index.js [hunk 3]
```diff
@@ -84,5 +84,8 @@ export const joinObjects = (objA, objB) => {
   isObject should also be set to true
 */
 export const addIsObjectProperty = (obj) => {
-  return obj['isObject'] = true; // I cant get this to work so I'm leaving it with what I thought should work
+  let newObj = {
+    isObject: true
+  };
+  return newObj;
```
Comment (matthewtsinontas) [inline objects/index.js]: Going through this line by line:
Line 87->89: I am creating a new object with a property isObject with a value of true
Line 90: Return new object

You are creating and returning a new object, you aren't modifying the existing one

Comment (stelisevil) [reply objects/index.js]: yeah thats why i wrote the extra tests. thought i mightve fucked that up!

Comment (matthewtsinontas) [reply objects/index.js]: Have another crack, im about with an eye on github so will be around

File: objects/index.js [hunk 4]
```diff
@@ -84,8 +84,8 @@ export const joinObjects = (objA, objB) => {
   isObject should also be set to true
 */
 export const addIsObjectProperty = (obj) => {
-  let newObj = {
+  const isObj = {
     isObject: true
   };
-  return newObj;
+  return Object.assign(obj, isObj);
```
Comment (matthewtsinontas) [inline objects/index.js]: Nice, this is actually a second way of solving this problem: 

```js
const newObj = obj;
newObj.isObject = true;
return newObj
```

Comment (matthewtsinontas) [reply objects/index.js]: We need to have a chat about whats going on with these functions too, the difference between modifying an existing object and creating a new one

**Pull request merged**